### ISSUE TITLE
Conv3D with padding='same' creates output shape error

### ISSUE THREAD

**Issue by kingjr** [keras-team/keras#9389]
Hi team, 

I'm encountering an error when adding a Dense layer on top of a Conv3D + MaxPool3D when `padding='same'`.

The error 
* is not generated during the creation of the model, but only when it is executed.
* disappears if either MaxPooling3D or Conv3D is removed.
* is present if `padding='same'` is used in either `MaxPooling3D` or `Conv3D`

```python
from keras.models import Sequential
from keras.layers import Conv3D, MaxPooling3D, Dense

shape = (5, 64, 64, 3)
x = np.zeros(np.r_[1, shape]) # input

for padding in ('valid', 'same'):
    model = Sequential()
    model.add(MaxPooling3D((1, 1, 1), padding=padding, input_shape=shape))
    model.add(Conv3D(128, 3, padding=padding))
    model.add(Dense(8))
    print(padding, model.predict(x).shape)
```

Do you what should be done?

(I'm using Keras 2.1.3)

**Comment by fchollet** [comment]
Your code works fine. My guess is that your default `image_data_format` in `~/.keras/keras.json` is set to "channels_first", whereas your data in your example assumes "channels_last". You can pass an explicit `data_format` argument to your layers to avoid this issue (or change the value in your config file).

**Comment by kingjr** [comment]
Thank @fchollet . Specifying `data_format` and/or changing `keras.json` didn't work, but upgrading tensorflow did.
